src/visibility.js

```diff
@@ -0,0 +1,170 @@
+// @flow
+import {BasePlugin, Utils} from 'playkit-js';
+import './style.css';
+import {DismissibleFloatingButtonComponent} from './components/dismissible/dismissible';
+
+const FLOATING_ACTIVE_CLASS: string = 'playkit-floating-active';
+const FLOATING_CONTAINER_CLASS: string = 'playkit-floating-container';
+const FLOATING_POSTER_CLASS: string = 'playkit-floating-poster';
+
+/**
+ * Visibility class.
+ * @classdesc
+ */
+class Visibility extends BasePlugin {
+  appTargetContainer: HTMLElement | null;
+  floatingContainer: HTMLElement;
+  floatingPoster: HTMLElement;
+  observer: window.IntersectionObserver;
+  everStartedPlaying: boolean = false;
+  dismissed: boolean = false;
+
+  /**
+   * The default configuration of the plugin.
+   * @type {Object}
+   * @static
+   */
+  static defaultConfig: Object = {
+    threshold: 0.5
+  };
+
+  getUIComponents() {
+    return [
+      {
+        label: 'dismissibleFloatingButtonComponent',
+        presets: ['Playback', 'Live'],
+        container: 'TopBarRightControls',
+        get: DismissibleFloatingButtonComponent,
+        props: {
+          onClose: () => {
+            this._dismissed();
+          }
+        }
+      }
+    ];
+  }
+
+  /**
+   * @static
+   * @public
+   * @returns {boolean} - Whether the plugin is valid.
+   */
+  static isValid(): boolean {
+    return true;
+  }
+
+  /**
+   * @constructor
+   * @param {string} name - The plugin name.
+   * @param {Player} player - The player instance.
+   * @param {Object} config - The plugin config.
+   */
+  constructor(name: string, player: Player, config: Object) {
+    super(name, player, config);
+    this.reset();
+    if (this.config.floating) {
+      this._initFloating();
+    }
+    const options = {
+      threshold: this.config.threshold
+    };
+
+    this.observer = new window.IntersectionObserver(this._handleVisibilityChange.bind(this), options);
+    this.observer.observe(this.appTargetContainer);
+  }
+
+  _initFloating() {
+    const defaultFloatingConfig = {
+      floating: {
+        position: 'bottom-right',
+        size: {
+          height: '225px',
+          width: '400px'
+        }
+      }
+    };
+    Utils.Object.mergeDeep(this.config, defaultFloatingConfig, Utils.Object.copyDeep(this.config));
+    this.floatingPoster = document.createElement('div');
+    this.floatingPoster.className = FLOATING_POSTER_CLASS;
+    this.floatingContainer = document.createElement('div');
+    this.floatingContainer.className = FLOATING_CONTAINER_CLASS;
+
+    // local const is defined to override flow null reference error
+    const appTargetContainer = (this.appTargetContainer = document.getElementById(this.player.config.targetId));
+    if (appTargetContainer) {
+      this.floatingContainer.innerHTML = appTargetContainer.innerHTML;
```

Comment:
Don't use innerHTML, it destroys and creates the DOM use `replaceChild` and `appendChild` which doesn't remove the DOM element, just re-references the DOM.